nanoFramework.Device.Bluetooth/BluetoothSignalStrengthFilter.cs

```diff
@@ -16,38 +19,187 @@ public class BluetoothSignalStrengthFilter
         private short _outOfRangeThresholdInDBm;
         private TimeSpan _outOfRangeTimeout;
 
+        Timer _scanCheck;
+        Hashtable _scanResults = new();
+        Object _scanResultsLock = new Object();
+
+        const int DEFAULT_DBM = -127;
+        const int DEFAULT_OOR_TIMEOUT = 60;
+
+        private class ScanItem
+        {
+            public short rssi;  // RSSI of last scan
+            public bool inRange; // True if device was in range.
+            public DateTime outRangeTime;  // Time device went out of range
+            public bool active;     // Scan item active, used for purging entries
+        }
+
         /// <summary>
         /// Create a new BluetoothSignalStrengthFilter object.
         /// </summary>
         public BluetoothSignalStrengthFilter()
         {
-            InRangeThresholdInDBm = -128;
-            OutOfRangeThresholdInDBm = -128;
-            OutOfRangeTimeout = new TimeSpan(0, 0, 10); // seconds
+            InRangeThresholdInDBm = DEFAULT_DBM;
+            OutOfRangeThresholdInDBm = DEFAULT_DBM;
+            OutOfRangeTimeout = new TimeSpan(0, 0, DEFAULT_OOR_TIMEOUT); // seconds
+
+            _scanResults = new Hashtable();
+
+            // Remove inactive entries from scan results every 1 minute 
+            TimeSpan time = new TimeSpan(0, 1, 0);
+            _scanCheck = new Timer(scanCheckCallback, 0, time, time);
         }
 
         ///// <summary>
-        ///// The interval at which received signal strength indicator (RSSI) events are sampled.
+        ///// The interval at which received signal strength indicator (RSSI) events are sampled.  (TODO)
         ///// </summary>
         //public TimeSpan? SamplingInterval { get; set; }
 
         /// <summary>
         /// The time out for a received signal strength indicator (RSSI) event to be considered
-        /// out of range.
+        /// out of range. Value between 1 and 60 seconds.
```

Comment:
if there is a range, where the range is checked? Native? If not, test it there and raise an exception if not in the range.
Similar comment for the next few ones.

Reply:
This whole BluetoothSignalStrengthFilter is going to be re-written in the next version. Didn't want to change too much for this PR, just wanted to fix the problem to existing version so as to clear issue.
I already realized the argument range checks were missing.

So I'll will leave as it is for the moment.